.github/workflows/publish-staging-redacted-export.yml

```
@@ -0,0 +1,140 @@
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install Node.js dependencies
+        run: yarn install
+
+      - name: Create schema artifact
+        env:
+          # NOTE: These services do not need to be running for this job
+          DATABASE_URL: postgres://postgres@localhost:5432
+          ELASTICSEARCH_URL: http://localhost:9200
+          REDIS_URL: redis://localhost:6379
+          RAILS_ENV: test
+          ANTIVIRUS_URL: http://localhost:3006/safe
+          ANTIVIRUS_USERNAME: av
+          ANTIVIRUS_PASSWORD: password
+          PSD_HOST: example.com
+          EMAIL_WHITELIST_ENABLED: false
+          TWO_FACTOR_AUTHENTICATION_ENABLED: false
+          CI: true
+        run: |
+          bin/rails redacted_export:generate_sql > redex/create_redacted_schema.sql
+
+      - name: Install cf client
+        env:
+          CF_CLI_VERSION: v7
+        run: |
+          mkdir -p $GITHUB_WORKSPACE/bin
+          curl -L "https://packages.cloudfoundry.org/stable?release=linux64-binary&version=${CF_CLI_VERSION}" | tar -zx -C $GITHUB_WORKSPACE/bin
+          echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
+
+      - name: Export database
+        id: export-database
+        env:
+          SPACE: staging
+        run: |
+          MAIN_DATABASE_SERVICE_NAME=psd-database
+          REDEX_ENV=psd-redex-env
+
+          TIMESTAMP=$(date +'%F-%H-%M-%S')
+          TEMP_DB_SERVICE_NAME=psd-redex-db-$TIMESTAMP
+          export TEMP_APP_NAME=psd-redex-app-$TIMESTAMP
+
+          echo "::set-output name=temp-db-service-name::$TEMP_DB_SERVICE_NAME"
+          echo "::set-output name=temp-app-name::$TEMP_APP_NAME"
+
+          cf api api.london.cloud.service.gov.uk
+          cf auth
+          cf target -o 'beis-opss' -s $SPACE
+
+          PSD_DATABASE_GUID=$(cf service $MAIN_DATABASE_SERVICE_NAME --guid)
+          PSD_DATABASE_PLAN=$(cf service $MAIN_DATABASE_SERVICE_NAME | awk -F ':' '$1=="plan" { gsub(/ /,"", $NF); print $NF }')
+          cf create-service postgres $PSD_DATABASE_PLAN $TEMP_DB_SERVICE_NAME -c "{\"restore_from_latest_snapshot_of\": \"$PSD_DATABASE_GUID\"}"
+          echo "::group::Waiting for db"
+          until cf service $TEMP_DB_SERVICE_NAME > /tmp/db_exists && grep -E "create succeeded|update succeeded" /tmp/db_exists; do sleep 30; echo "Waiting for db..."; done
+          echo "::endgroup::"
+
+          cf push $TEMP_APP_NAME --task -f redex/manifest.yml --var app-name=$TEMP_APP_NAME --var db-name=$TEMP_DB_SERVICE_NAME --var env-name=$REDEX_ENV
+
+          cf run-task $TEMP_APP_NAME
+          echo "::group::Waiting for task"
+          timeout 15m bash -c 'until cf tasks $TEMP_APP_NAME > /tmp/task_result && grep -E "SUCCEEDED" /tmp/task_result; do sleep 10; echo "Waiting for task..."; done'
+          echo "::endgroup::"
+
+          cf delete-service $TEMP_APP_NAME -f
```

I'm not sure `delete-service will work for an application. You might need:
```suggestion
          cf delete $TEMP_APP_NAME -f
```

---

👍 https://github.com/UKGovernmentBEIS/beis-opss-psd/pull/1580/commits/404e52a5710deea55b15a1218938fd8839da9ba5